File: src/Monitoring.Web/Monitoring.Business/Service/DeviceService.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using JetBrains.Annotations;
using Monitoring.Business.Abstract.Repository;
using Monitoring.Business.Model;

namespace Monitoring.Business.Service
{
  [UsedImplicitly]
  public class DeviceService
  {
    private readonly IDeviceRepository _deviceRepository;

    public DeviceService(IDeviceRepository deviceRepository)
    {
      _deviceRepository = deviceRepository;
    }

    public Task<Device> GetDevice(string deviceUuid)
    {
      return _deviceRepository.GetDevice(deviceUuid);
    }

    public async Task<IEnumerable<Device>> GetDevices(bool isActive)
    {
      var devices = await _deviceRepository.GetDevices();

      return devices.Where(x => !isActive || x.IsActive);
    }
  }
}

File: src/Monitoring.Client/src/app/devices/services/device.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { Device } from '../models';

@Injectable()
export class DeviceService {
  constructor(private httpClient: HttpClient) {}

  getDevices(isActive: boolean): Observable<Device[]> {
    return this.httpClient.get<Device[]>(`/devices?active=${isActive}`);
  }

  getDevice(deviceUuid: string): Observable<Device> {
    return this.httpClient.get<Device>(`/devices/${deviceUuid}`);
  }
}
